pkg/logs/pipeline/pipeline.go

```diff
@@ -17,36 +19,37 @@ import (
 type Pipeline struct {
 	InputChan chan *message.Message
 	processor *processor.Processor
-	sender    *sender.Sender
+	sender    sender.Sender
 }
 
 // NewPipeline returns a new Pipeline
-func NewPipeline(outputChan chan *message.Message, processingRules []*config.ProcessingRule, endpoints *client.Endpoints, destinationsContext *client.DestinationsContext) *Pipeline {
-	// initialize the main destination
-	main := client.NewDestination(endpoints.Main, destinationsContext)
-
-	// initialize the additional destinations
-	var additionals []*client.Destination
-	for _, endpoint := range endpoints.Additionals {
-		additionals = append(additionals, client.NewDestination(endpoint, destinationsContext))
-	}
-
+func NewPipeline(outputChan chan *message.Message, processingRules []*config.ProcessingRule, endpoints *config.Endpoints, destinationsContext *tcp.DestinationsContext) *Pipeline {
 	// initialize the sender
-	destinations := client.NewDestinations(main, additionals)
+	destinations := client.NewDestinations(endpoints, destinationsContext)
 	senderChan := make(chan *message.Message, config.ChanSize)
-	sender := sender.NewSender(senderChan, outputChan, destinations)
+	var newsender sender.Sender
+	if coreConfig.Datadog.GetBool("logs_config.use_http") {
+		newsender = sender.NewBatchSender(senderChan, outputChan, destinations)
+	} else {
+		newsender = sender.NewSingleSender(senderChan, outputChan, destinations)
+	}
 
 	// initialize the input chan
 	inputChan := make(chan *message.Message, config.ChanSize)
 
 	// initialize the processor
-	encoder := processor.NewEncoder(endpoints.Main.UseProto)
+	var encoder processor.Encoder
+	if coreConfig.Datadog.GetBool("logs_config.use_http") {
+		encoder = processor.NewHTTPEncoder()
+	} else {
+		encoder = processor.NewTCPEncoder(endpoints.Main.UseProto)
```

Comment:
For this one, we should have ProtoEncoder or RawEncoder depending on use_proto config flag.